modules/local/validation/metadata/resources/usr/bin/main.py

```diff
@@ -0,0 +1,173 @@
+        if not analysis_type and 'analysisType' in payload:
+            analysis_type = payload['analysisType'].get('name', '')
+        
+        print(f'Analysis type: {analysis_type or "unknown"}')
+        
+        # Check if read group is required (only for sequenceExperiment)
+        read_group_required = analysis_type == "sequenceExperiment"
+        
+        # Load biospecimen data
+        specimens = load_biospecimen_data(specimen_file, 'specimen')
+        samples = load_biospecimen_data(sample_file, 'sample')
+        experiments = load_biospecimen_data(experiment_file, 'experiment')
+        
+        # Load read groups if required or if file is provided
+        read_groups = []
+        if read_group_required or (read_group_file and os.path.exists(read_group_file) and os.path.getsize(read_group_file) > 0):
+            read_groups = load_biospecimen_data(read_group_file, 'read_group')
+        
+        # Validate read group requirement
+        if read_group_required and not read_groups:
+            raise ValueError(f'Read group data is required for analysis type "{analysis_type}" but no valid read group data found')
+        
+        print(f'Read group required: {read_group_required}')
+        print(f'Loaded biospecimen data: {len(specimens)} specimens, {len(samples)} samples, {len(experiments)} experiments, {len(read_groups)} read groups')
+        
+        # Perform cross-validation checks (individual validation done upstream)
+        validation_errors = []
+        
+        # Cross-validate read group file references with payload (if read groups exist)
+        if read_groups:
+            validation_errors.extend(validate_read_group_files(payload, read_groups))
+        
+        if validation_errors:
+            return False, validation_errors
+        
+        print('Payload-read group file consistency validation completed successfully')
+        return True, []
+    
+    except Exception as e:
+        # All exceptions from comprehensive_validation are sent to stderr
+        return False, [f'Unexpected validation error: {str(e)}']
+
+
+def main():
+    """Main function for command line execution."""
+    parser = argparse.ArgumentParser(description='Cross-validate file consistency between payload and read group data (assumes upstream validation of individual files and relationships)')
+    parser.add_argument('payload_file', help='Path to analysis payload JSON file')
+    parser.add_argument('--specimen-file', required=True, help='Path to specimen TSV file')
+    parser.add_argument('--sample-file', required=True, help='Path to sample TSV file')
+    parser.add_argument('--experiment-file', required=True, help='Path to experiment TSV file')
+    parser.add_argument('--read-group-file', help='Path to read group TSV file (analysis type dependent)')
+    parser.add_argument('--analysis-type', help='Analysis type from metadata (optional, will be extracted from payload if available)')
```

Comment:
think we should make analysis-type required. Even if other earlier scripts have already checked for this, checking again for sanity.